### PR TITLE
feat: adds octuples modal component

### PR DESCRIPTION
**Pull request by ychhabra-eightfold**

### CODE REVIEW

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    modalClassName,
+    modalWrapperClassName,
+    ...rest
+}) => {
+    const modalClasses: string = classNames([
+        styles.modal,
+        modalClassName,
+        { [styles.small]: size === ModalSize.small },
+        { [styles.medium]: size === ModalSize.medium },
+        { [styles.large]: size === ModalSize.large },
```
Comment (dylankilgore) [inline src/components/Modal/Modal.tsx:22]: Need Fullscreen modal too, there are many scenarios and existing features in the product that require this.

File: src/components/Modal/Modal.types.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { BaseDialogProps } from '../Dialog/BaseDialog/BaseDialog.types';
+
+export enum ModalSize {
+    small = 'small',
+    medium = 'medium',
+    large = 'large',
```
Comment (dylankilgore) [inline src/components/Modal/Modal.types.ts:6]: fullscreen

**Review by dylankilgore — CHANGES_REQUESTED** [review]

File: src/components/Dialog/BaseDialog/BaseDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            <div
+                className={dialogClasses}
+                style={dialogStyle}
+                onClick={stopPropagation}
+            >
+                <div className={headerClasses}>
+                    <span id={labelId}>{header}</span>
+                    <NeutralButton icon={IconName.mdiClose} onClick={onClose} />
+                </div>
+                <div className={bodyClassName}>{body}</div>
```
Comment (dylankilgore) [inline src/components/Dialog/BaseDialog/BaseDialog.tsx:73]: To make this scrollable you may consider adding a bodyMaxHeight prop. then refer to line 57 in the Modal.js common component in vscode `www/react/src/lib/common/Modal.js`. We either take the specified height or calculate it using the viewport, taking into account any variable insets and heights of the header and footer.

Comment (dylankilgore) [reply src/components/Dialog/BaseDialog/BaseDialog.tsx:73]: You may also look at line 136 of the Modal common component scss to see the default calc for some ideas --> `www/react/src/styles/Modal.scss`

**Review by dylankilgore — CHANGES_REQUESTED** [review]

**Review by dkilgore-eightfold — APPROVED** [review]

**Pull request merged**